pkg/models/notification/notification.go

```diff
@@ -0,0 +1,383 @@
+
+	// If user is nil, it will list all global object.
+	if user == "" {
+		appendGlobalLabel(resource, q)
+	} else {
+		// If the user is not nil, only return the object belong to this user.
+		appendTenantLabel(user, q)
+	}
+
+	return o.resourceGetter.List(resource, "", q)
+}
+
+// Get the specified object, if you want to get a global object, the user must be nil.
+// If you want to get a tenant object, the user must equal to the tenant specified in labels of the object.
+func (o *operator) Get(user, resource, name string) (runtime.Object, error) {
+	obj, err := o.resourceGetter.Get(resource, "", name)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := authorizer(user, obj); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+// Create a object, if the user is nil, it only can create a global object.
+// If the user is not nil, it only can create a tenant object.
+func (o *operator) Create(user, resource string, obj runtime.Object) (runtime.Object, error) {
+
+	if err := authorizer(user, obj); err != nil {
+		return nil, err
+	}
+
+	switch resource {
+	case v2.ResourcesPluralDingTalkConfig:
+		return o.ksClient.NotificationV2().DingTalkConfigs().Create(context.Background(), obj.(*v2.DingTalkConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralDingTalkReceiver:
+		return o.ksClient.NotificationV2().DingTalkReceivers().Create(context.Background(), obj.(*v2.DingTalkReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralEmailConfig:
+		return o.ksClient.NotificationV2().EmailConfigs().Create(context.Background(), obj.(*v2.EmailConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralEmailReceiver:
+		return o.ksClient.NotificationV2().EmailReceivers().Create(context.Background(), obj.(*v2.EmailReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralSlackConfig:
+		return o.ksClient.NotificationV2().SlackConfigs().Create(context.Background(), obj.(*v2.SlackConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralSlackReceiver:
+		return o.ksClient.NotificationV2().SlackReceivers().Create(context.Background(), obj.(*v2.SlackReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralWebhookConfig:
+		return o.ksClient.NotificationV2().WebhookConfigs().Create(context.Background(), obj.(*v2.WebhookConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralWebhookReceiver:
+		return o.ksClient.NotificationV2().WebhookReceivers().Create(context.Background(), obj.(*v2.WebhookReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralWechatConfig:
+		return o.ksClient.NotificationV2().WechatConfigs().Create(context.Background(), obj.(*v2.WechatConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralWechatReceiver:
+		return o.ksClient.NotificationV2().WechatReceivers().Create(context.Background(), obj.(*v2.WechatReceiver), v1.CreateOptions{})
+	default:
+		return nil, errors.NewInternalError(nil)
+	}
+}
+
+// Delete a object, if the user is nil, it only can delete the global object.
```

Comment:
// Delete a object, if the user is nil, it only can delete the global object.
// If the user is not nil, it only can delete the tenant object which the tenant
// specified in the labels of the object equal to this user.
=>
// Delete an object. A global object will be deleted if the user is nil.
// If the user is not nil, a tenant object whose tenant label matches the user will be deleted.